services/consensuscontext/create_block.go

```diff
@@ -10,26 +10,25 @@ import (
 
 func (s *service) createTransactionsBlock(blockHeight primitives.BlockHeight, prevBlockHash primitives.Sha256) (*protocol.TransactionsBlockContainer, error) {
 
-	proposedTransactions, err := s.transactionPool.GetTransactionsForOrdering(&services.GetTransactionsForOrderingInput{
+	proposedTransactions, err := s.fetchTransactions(&services.GetTransactionsForOrderingInput{
```

Comment:
since you introduced a private method, might as well move the ugly inter-service struct `services.GetTransactionsForOrderingInput` into it and have it take an `int` argument

Reply:
done